workflows/src/main/kotlin/com/r3/corda/lib/ci/workflows/SignedKeyUtilities.kt

```diff
@@ -76,13 +76,13 @@ private fun ServiceHub.concatChallengeResponseAndSign(
  */
 @CordaInternal
 @VisibleForTesting
-fun verifySignedChallengeResponseSignature(signedKeyForAccount: SignedKeyForAccount) {
+fun verifySignedChallengeResponseSignature(signedKeyForAccount: SignedKeyForAccount, knownKey: PublicKey?) {
     try {
         signedKeyForAccount.signedChallengeResponse.sig.verify(signedKeyForAccount.signedChallengeResponse.raw.hash.bytes)
     } catch (ex: SignatureException) {
         throw SignatureException("The signature on the object does not match that of the expected public key signature", ex)
     }
-    if (signedKeyForAccount.signedChallengeResponse.sig.by != signedKeyForAccount.publicKey) {
+    if (signedKeyForAccount.signedChallengeResponse.sig.by != knownKey?: signedKeyForAccount.publicKey ) {
```

Comment:
Can this check be put before the signature verification since this will fail faster than the verification?